lib/scripting/objects/vpm-controller.ts

```diff
@@ -18,16 +19,294 @@
+		}
+	}
+	public Stop(): void {
+		logger().debug('STOP');
+		//TODO unclear what to do
+	}
+
+	// GameSetting
+	// NOTE: Dip - implemented using Proxy
+
+	get HandleMechanics(): number {
+		return 0;
+	}
+	set HandleMechanics(mechanicNr: number) {
+		logger().debug('TODO HandleMechanics');
+	}
+	/**
+	 * Determine if game uses WPC Numbering of Switches and Lamps
+	 * WPCNumbering = Column*10 + Row (11,12,13,14,15,16,17,18,21,22...)
+	 * non WPCnumbering = 1,2,3,4,...
+	 */
+	get WPCNumbering(): number {
+		logger().debug('WPCNumbering');
+		return 1;
+	}
+	get SampleRate(): number {
+		logger().debug('SampleRate');
+		return 22050;
+	}
+
+	//Customization
+	get SplashInfoLine(): string {
+		return this.splashInfoLine;
+	}
+	set SplashInfoLine(gameCredits) {
+		this.splashInfoLine = gameCredits;
+	}
+	get ShowFrame(): boolean {
+		return false;
+	}
+	set ShowFrame(showFrame: boolean) {
+		logger().debug('ShowFrame', showFrame);
+	}
+	get DoubleSize(): boolean {
+		return false;
+	}
+	set DoubleSize(doubleSize: boolean) {
+		logger().debug('DoubleSize', doubleSize);
+	}
+	get Antialias(): boolean {
+		return false;
+	}
+	set Antialias(enabled: boolean) {
+		logger().debug('Antialias', enabled);
+	}
+	get BorderSizeX(): number {
+		return 0;
+	}
+	set BorderSizeX(size: number) {
+		logger().debug('BorderSizeX', size);
+	}
+	get BorderSizeY(): number {
+		return 0;
+	}
+	set BorderSizeY(size: number) {
+		logger().debug('BorderSizeY', size);
+	}
+	get WindowPosX(): number {
+		return 0;
+	}
+	set WindowPosX(position: number) {
+		logger().debug('WindowPosX', position);
+	}
+	get WindowPosY(): number {
+		return 0;
+	}
+	set WindowPosY(position: number) {
+		logger().debug('WindowPosY', position);
+	}
+	get LockDisplay(): boolean {
+		return false;
+	}
+	set LockDisplay(locked: boolean) {
+		logger().debug('LockDisplay', locked);
+	}
+	get Hidden(): boolean {
+		return false;
+	}
+	set Hidden(hidden: boolean) {
+		logger().debug('Hidden', hidden);
+	}
+	public SetDisplayPosition(x: number, y: number, hWnd: any): void {
+		logger().debug('SetDisplayPosition', { x, y, hWnd });
+	}
+	public ShowOptsDialog(hWnd: any): void {
+		logger().debug('ShowOptsDialog', hWnd);
+	}
+	public ShowPathesDialog(hWnd: any): void {
+		logger().debug('ShowPathesDialog', hWnd);
+	}
+	public ShowAboutDialog(hWnd: any): void {
+		logger().debug('ShowAboutDialog', hWnd);
+	}
+	/**
+	 * Checks the rom set for the current game and displays the results.
+	 * @param nShowOptions: 0 = Always displays the results, 1 = Only displays the results if there are errors found, 2 = Never displays the results
+	 * @returns true if the roms are good.
+	 */
+	public CheckROMS(nShowOptions: number): boolean {
+		logger().debug('CheckROMS', nShowOptions);
+		return true;
 	}
 
-	public Run(): void {
+	// AggregatePollingFunctions
+	get ChangedLamps(): number[][] {
+		const changedLamps: number[][] = this.emulator.emulatorState.getChangedLamps();
+		return changedLamps;
```

Comment:
Can return this directly now

Reply:
yes - reason is i add a debug call in between when i "debug"

Reply:
Fair enough :)